hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestDirectoryDeletingServiceV1.java

```diff
@@ -0,0 +1,273 @@
+    fs.delete(appRoot, true);
+
+
+    // After Delete
+    checkPath(appRoot);
+
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(0, deletedDirTable),
+        1000, 120000); // 2 minutes
+
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(1, dirTable),
+        1000, 120000); // 2 minutes
+
+    Assert.assertTrue(dirTable.iterator().hasNext());
+    Assert.assertEquals(root.getName(),
+        dirTable.iterator().next().getValue().getName());
+
+    DirectoryDeletingService dirDeletingService =
+        (DirectoryDeletingService) cluster.getOzoneManager().getKeyManager()
+            .getDirDeletingService();
+
+    GenericTestUtils.waitFor(
+        () -> dirDeletingService.getDeletedDirsCount() >= 1,
+        1000, 10000);
+    GenericTestUtils.waitFor(
+        () -> dirDeletingService.getMovedFilesCount() >= 0,
+        1000, 10000);
+
+    Assert.assertTrue(dirDeletingService.getRunCount() > 1);
+  }
+
+  @Test
+  public void testDeleteWithMultiLevels() throws Exception {
+    Path root = new Path("/rootDir");
+    Path appRoot = new Path(root, "appRoot");
+
+    for (int i = 1; i <= 3; i++) {
+      Path parent = new Path(appRoot, "parentDir" + i);
+      Path child = new Path(parent, "childFile");
+      ContractTestUtils.touch(fs, child);
+    }
+
+    Table<String, OmKeyInfo> deletedDirTable =
+        cluster.getOzoneManager().getMetadataManager().getDeletedDirTable();
+    Table<String, OmKeyInfo> keyTable =
+        cluster.getOzoneManager().getMetadataManager().getKeyTable();
+    Table<String, OmDirectoryInfo> dirTable =
+        cluster.getOzoneManager().getMetadataManager().getDirectoryTable();
+
+    // Before delete
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(0, deletedDirTable),
+        1000, 120000); // 2 minutes
+
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(3, keyTable),
+        1000, 120000); // 2 minutes
+
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(5, dirTable),
+        1000, 120000); // 2 minutes
+
+    // Delete the rootDir, which should delete all keys.
+    fs.delete(root, true);
+
+    // After Delete
+    checkPath(root);
+
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(0, deletedDirTable),
+        1000, 120000); // 2 minutes
+
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(0, keyTable),
+        1000, 120000); // 2 minutes
+
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(0, dirTable),
+        1000, 120000); // 2 minutes
+
+    DirectoryDeletingService dirDeletingService =
+        (DirectoryDeletingService) cluster.getOzoneManager().getKeyManager()
+            .getDirDeletingService();
+
+    GenericTestUtils.waitFor(
+        () -> dirDeletingService.getDeletedDirsCount() >= 5,
+        1000, 10000);
+    GenericTestUtils.waitFor(
+        () -> dirDeletingService.getMovedFilesCount() >= 3,
+        1000, 10000);
```

Comment:
Can we record the getDeletedDirsCount/getMovedFilesCount before the delete, and then get these value again and compare the delta value. This way would be the more accurate check.

Reply:
Sure, will do it.

Reply:
Done!